src/ashley/api/serializers.py

```diff
@@ -0,0 +1,45 @@
+"""Serializer for ashley api."""
+from django.contrib.auth import get_user_model
+from rest_framework import serializers
+
+from ashley.context_mixins import get_current_lti_session
+from ashley.defaults import (
+    _FORUM_ROLE_INSTRUCTOR,
+    _FORUM_ROLE_MODERATOR,
+    _FORUM_ROLE_STUDENT,
+)
+
+User = get_user_model()
+
+
+class UserSerializer(serializers.ModelSerializer):
+    """Serializer for User model."""
+
+    role = serializers.SerializerMethodField()
+
+    class Meta:
+        model = User
+        fields = ["id", "public_username", "role"]
+        read_only_fields = ["public_username", "role"]
+
+    def get_role(self, obj):
+        """Describes the role of the user"""
+        context = get_current_lti_session(self.context.get("request"))
+        if obj.is_active:
+            if obj.groups.filter(
```

Comment:
Should be great to do it in one queryset, maybe something like this ? (`Q` must be imported first `from django.db.models import Q`)

```suggestion
            if obj.groups.filter(name=context.get_group_role_name(_FORUM_ROLE_STUDENT), ~Q(name=context.get_group_role_name(_FORUM_ROLE_MODERATOR)))
```

Reply:
didn't manage to do it with Q or with combined filters, I think it's because I'm checking on many to many values, anyway i changed the code to only get one query by getting once the group list and just checking if group is part of it.